### PR TITLE
TITAN-1234: Set $jwtap_jwt_payload

### PR DESCRIPTION
**Pull request by DaveWillkomm**
Set $jwtap_jwt_payload
=================
* Set $jwtap_jwt_payload for use by subsequent access handlers
* Updated ngx_mruby
* Refactored specs
* Updated testing gems

Refactor access handler as a class
=========================
* This is in preparation of supporting multiple scripts per directive
* Removed $jwtap_algorithm while mruby-jwt only supports HS256
* Default $jwtap_cookie_name to "jwt"
* Default $jwtap_cookie_expiration_duration_seconds to 1800

Support multiple scripts per directive
===========================
* As ngx_mruby does not yet support this (see matsumotory/ngx_mruby#128), implemented mruby_directive_paths.rb
* Spec options is an OpenStruct, so don't need to use Hash syntax ;)
* Renamed spec/functional/lib/jwtap/access_handler_spec.rb as lib/jwtap_spec.rb

### CODE REVIEW

File: nginx/conf/nginx.conf.erb [hunk 1]
```diff
@@ -58,7 +66,7 @@ http {
 
         location /set-cookie {
           mruby_content_handler_code '
-            var = Nginx::Var.new
+            var = Nginx::Request.new.var
             expiration = (Time.now + var.jwtap_expiration_duration_seconds.to_i).to_i
             jwt = JWT.encode({ "exp" => expiration }, Base64.decode(var.jwtap_secret_key_base64), var.jwtap_algorithm)
```
Comment (stlcardsfan) [inline nginx/conf/nginx.conf.erb]: var.jwtap_algorithm is no longer being set, correct?

Comment (DaveWillkomm) [inline nginx/conf/nginx.conf.erb]: Correct! I mentioned why in the MR description, "Removed $jwtap_algorithm while mruby-jwt only supports HS256".

Comment (stlcardsfan) [inline nginx/conf/nginx.conf.erb]: Cool, I noticed that.  Should that line be updated to remove the reference to var.jwtap_algorithm?

Comment (DaveWillkomm) [inline nginx/conf/nginx.conf.erb]: Oh, now I see what you're getting at. Lemme take a closer look.

**Comment by DaveWillkomm** [comment]
@stlcardsfan Added a commit to cleanup that old variable - thanks for catching it.

**Pull request merged**